packages/@uppy/transloadit/src/AssemblyOptions.js

```diff
@@ -65,7 +65,9 @@ class AssemblyOptions {
     if (file == null) return undefined
 
     const options = this.opts
-    const assemblyOptions = await options.getAssemblyOptions(file, options)
+    const assemblyOptions = typeof options.assemblyOptions === 'function'
```

Comment:
isn't this a breaking change? shouldn't we test for *both* `getAssemblyOptions` and `assemblyOptions` in order to maintain bakcwards compatibility?

Reply:
This is `AssemblyOptions`, which is passed `opts.assemblyOptions`, which is:

```js
this.opts.assemblyOptions = opts.assemblyOptions ?? this.opts.assemblyOptions
```

Reply:
If the user is passing only `fields`, how does that work? Shouldn't we add TODO comments to remember to remove that logic when we remove support for the deprecated fields?

Reply:
It's all quite confusing at the moment, it's not clear what options supersede what if you pass them directly or in `assemblyOptions`. But nothing on that front changed in this PR, everything still works in the combinations people did before, there is now just another option.

Regarding removing that logic next major, I think it's obvious enough that all references should be removed and only `assemblyOptions` should remain?

Reply:
> I think it's obvious enough

Famous last words ;)